doc/ssdmond/stormond-hld.md

```diff
@@ -0,0 +1,211 @@
+
+#### **2.4.1 SsdBase API additions**
+
+In order to collect IO reads/writes and number of reserved blocks, we would need to add the following member methods to the `SsdBase` class in [ssd_base.py](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_ssd/ssd_base.py) and provide a generic implementation in [ssd_generic.py](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_ssd/ssd_generic.py):
+
+
+```
+class SsdBase(object):
+
+...
+
+def get_io_reads(self):
+"""
+Retrieves the total number of Input/Output (I/O) reads done on an SSD
+
+Returns:
+    An integer value of the total number of I/O reads
+"""
+
+def get_io_writes(self):
+"""
+Retrieves the total number of Input/Output (I/O) writes done on an SSD
+
+Returns:
+    An integer value of the total number of I/O writes
+"""
+
+def get_reserves_blocks(self):
+"""
+Retrieves the total number of reserved blocks in an SSD
+
+Returns:
+    An integer value of the total number of reserved blocks
+"""
+
+```
+
+#### **2.4.2 Support for Multiple Storage Disks**
+
+The `ssdutil` utility assumes that the disk drive is `/dev/sda` whereas the drive letter could be any label based on the number of SSDs. It could also be a different type of storage device such as eMMC, eUSB or NVMe.
+
+In order to get a clear picture of the number and type of disks present on a device, we introduce a new class `StorageDevices()` which will accompany the stormond daemon at `sonic-platform-daemons/sonic-stormond/scripts/StorageDevices.py`. This new class provides the following methods:
+
+```
+class StorageDevices():
+
+# A dictionary where the key is the name of the disk and the value is the corresponding class object
+devices = {}
+
+...
+
+def get_storage_devices(self):
+"""
+Retrieves all the storage disks on the device and adds their names as key to the 'devices' dict.
+"""
+
+def get_storage_device_object(self):
+"""
+Instantiates an object of the corresponding storage device class. 
+Adds the instantiated class object as a value to the corresponding key in the dictionary object.
+
+NOTE: SsdUtil is supported currently. Future support for EmmcUtil, eUSBUtil and NVMeUtil
+
+"""
+
+```
+
+This class is a helper to the Storage Daemon class. 
+
+Example usage:
+
+Assuming a device contains the following storage disks:
+```
+root@str-a7280cr3-2:~# ls /sys/block/
+loop0  loop1  loop2  loop3  loop4  loop5  loop6  loop7  **mmcblk0**  mmcblk0boot0  mmcblk0boot1  **sda**
+```
+
+We would instantiate an object of the StorageDevices() class
+`storage = StorageDevices()`
+
+`storage.devices` would contain `{"mmcblk0": <Emmcutil object>, "sda": <SsdUtil object>}`
+
+we would then get static and dynamic information by leveraging the respective member function implementations of `SsdUtil` and `EmmcUtil`, as they both derive from `SsdBase`.
+
+**NOTE:** eUSBUtil and NVMeUtil classes are not yet available.
```

Comment:
Echo to my previous comment where we could just have `UsbUtil` here instead.

Reply:
Updated in latest commit